### PR TITLE
Cloning v1beta3 as v1 and exposing it in the apiserver

### PR DESCRIPTION
**Pull request by nikhiljindal**
For https://github.com/GoogleCloudPlatform/kubernetes/issues/7018

### CODE REVIEW

**Comment by nikhiljindal** [comment]
cc @bgrant0607

File: hack/local-up-cluster.sh [hunk 1]
```diff
@@ -124,7 +124,7 @@ sudo -E "${GO_OUT}/kube-apiserver" \
   --admission_control="${ADMISSION_CONTROL}" \
   --address="${API_HOST}" \
   --port="${API_PORT}" \
-  --runtime_config=api/v1beta3 \
+  --runtime_config=api/v1 \
```
Comment (bgrant0607) [inline hack/local-up-cluster.sh]: We shouldn't enable v1 by default.

File: pkg/api/v1/conversion.go [hunk 2]
```diff
@@ -0,0 +1,2810 @@
+/*
+Copyright 2015 Google Inc. All rights reserved.
```
Comment (bgrant0607) [inline pkg/api/v1/conversion.go]: Why do we need all these conversions? I'd expect the internal rep to be very close to the v1 API.

File: pkg/api/v1/register.go [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+		&DeleteOptions{},
+		&ListOptions{},
+		&PodLogOptions{},
+		&PodExecOptions{},
+		&PodProxyOptions{},
+		&ComponentStatus{},
+		&ComponentStatusList{},
+	)
+	// Legacy names are supported
+	api.Scheme.AddKnownTypeWithName("v1", "Minion", &Node{})
```
Comment (bgrant0607) [inline pkg/api/v1/register.go]: We shouldn't support these legacy names.

**Comment by bgrant0607** [comment]
cc @smarterclayton @lavalamp

File: pkg/api/v1/conversion.go (hunk 2 above)
Comment (lavalamp) [inline pkg/api/v1/conversion.go]: It's an optimization that @wojtek-t is working on. They're auto-generated.

**Comment by lavalamp** [comment]
Looks OK, unfortunate that everything has to be done in four places now...

**Comment by bgrant0607** [comment]
Hopefully we can get rid of v1beta1/2 in a couple weeks, which should help a lot.

**Comment by bgrant0607** [comment]
LGTM. Will merge on green. We can fix anything necessary, like removing minions, in subsequent, smaller PRs.

**Comment by bgrant0607** [comment]
Both travis and shippable are very backed up, so it may be a while.

**Comment by nikhiljindal** [comment]
Yes. Right now, this is just an "as-is" copy of v1beta3.

File: hack/local-up-cluster.sh (hunk 1 above)
Comment (nikhiljindal) [inline hack/local-up-cluster.sh]: Reverted

**Comment by nikhiljindal** [comment]
Shippable is green!!

**Pull request merged**